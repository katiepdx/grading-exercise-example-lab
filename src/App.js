import { useState } from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  const [message, setMessage] = useState('');

  const handleClick = () => {
    setMessage('Hello!!!');
    setTimeout(() => setMessage(''), 1000);
  };

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />

        <p aria-label="message">{message}</p>

        <p>
          <button aria-label="view-message" onClick={handleClick}>View Message</button>
        </p>
      </header>
    </div>
  );
}

export default App;
