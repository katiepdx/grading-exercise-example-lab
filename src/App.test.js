import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

test('displays hello message when button is clicked', () => {
  render(<App />);
  const button = screen.getByRole('button', { name: 'view-message' });
  fireEvent.click(button);

  const message = screen.getByText('Hello!!!');

  expect(message).toBeInTheDocument();
  expect(message).toMatchSnapshot();
});
