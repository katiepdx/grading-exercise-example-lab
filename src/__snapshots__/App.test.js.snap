// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays hello message when button is clicked 1`] = `
<p
  aria-label="message"
>
  Hello!!!
</p>
`;
